File: frontend/src/modules/orcamentos/utils/investimentoDescricao.ts
export const INVESTIMENTO_DESCRICAO_DEMAIS_PADRAO = 'Demais itens da proposta'

export function descricaoInvestimentoConsolidadaPadrao(titulo: string): string {
  const t = titulo.trim()
  return t
    ? `Solução completa conforme escopo técnico - ${t}`
    : 'Solução completa conforme escopo técnico'
}

/** Texto exibido na tabela de investimento (custom ou padrão). */
export function descricaoInvestimentoExibicao(
  custom: string | null | undefined,
  padrao: string
): string {
  const valor = (custom ?? '').trim()
  return valor || padrao
}

export function placeholderDescricaoInvestimento(titulo: string): string {
  return INVESTIMENTO_DESCRICAO_DEMAIS_PADRAO
}

File: backend/apps/orcamentos/services/investimento_descricao.py
"""Textos padrão da descrição de investimento (solução completa)."""
from __future__ import annotations

INVESTIMENTO_DESCRICAO_DEMAIS_PADRAO = "Demais itens da proposta"


def descricao_investimento_consolidada_padrao(titulo: str) -> str:
    titulo = (titulo or "").strip()
    if titulo:
        return f"Solução completa conforme escopo técnico - {titulo}"
    return "Solução completa conforme escopo técnico"


def descricao_investimento_exibicao(orcamento, padrao: str) -> str:
    custom = (getattr(orcamento, "investimento_descricao", None) or "").strip()
    return custom or padrao
